client.go

```diff
@@ -1326,79 +1292,43 @@ func (c *Client) refreshCmd(cmd *protocol.RefreshRequest) (*clientproto.RefreshR
 
 	resp := &clientproto.RefreshResponse{}
 
-	token := cmd.Token
-	if token == "" {
+	if cmd.Token == "" {
 		c.node.logger.log(newLogEntry(LogLevelInfo, "client token required to refresh", map[string]interface{}{"user": c.user, "client": c.uid}))
 		return resp, DisconnectBadRequest
 	}
 
 	config := c.node.Config()
 
-	var user string
-	var expireAt int64
-	var info Raw
-	var b64info string
-
-	if cmd.Token == "" {
-		c.node.logger.log(newLogEntry(LogLevelInfo, "refresh token required", map[string]interface{}{"client": c.uid, "user": c.UserID()}))
-		return resp, DisconnectInvalidToken
-	}
-	parsedToken, err := jwt.ParseWithClaims(token, &connectTokenClaims{}, jwtKeyFunc(config))
-	if parsedToken == nil && err != nil {
-		c.node.logger.log(newLogEntry(LogLevelInfo, "invalid refresh token", map[string]interface{}{"error": err.Error(), "client": c.uid, "user": c.UserID()}))
-		return resp, DisconnectInvalidToken
-	}
-	if claims, ok := parsedToken.Claims.(*connectTokenClaims); ok && parsedToken.Valid {
-		user = claims.StandardClaims.Subject
-		info = claims.Info
-		b64info = claims.Base64Info
-		expireAt = claims.StandardClaims.ExpiresAt
-	} else {
-		if validationErr, ok := err.(*jwt.ValidationError); ok {
-			if validationErr.Errors == jwt.ValidationErrorExpired {
-				// The only problem with token is its expiration - no other errors set in bitfield.
-				resp.Error = ErrorTokenExpired
-				return resp, nil
-			}
-			c.node.logger.log(newLogEntry(LogLevelInfo, "invalid refresh token", map[string]interface{}{"error": err.Error(), "client": c.uid, "user": c.UserID()}))
-			return resp, DisconnectInvalidToken
-		}
-		c.node.logger.log(newLogEntry(LogLevelInfo, "invalid refresh token", map[string]interface{}{"error": err.Error(), "client": c.uid, "user": c.UserID()}))
-		return resp, DisconnectInvalidToken
+	var (
+		token     connectToken
+		errVerify error
+	)
+	if token, errVerify = c.node.verifyConnectToken(cmd.Token); errVerify != nil {
```

Comment:
This behaviour does not mimic previous one, no proper token verify error handling. Seems that this is also true for other places